src/main/java/appeng/me/storage/CellInventory.java

```diff
@@ -444,16 +446,62 @@ protected void loadCellItems()
 
 		for( int x = 0; x < types; x++ )
 		{
-			final ItemStack t = ItemStack.loadItemStackFromNBT( this.tagCompound.getCompoundTag( itemSlots[x] ) );
-			if( t != null )
+			// Get the item NBT, which can crash if someone daft stored something other than a compound NBT with the same key in our cell
+			final NBTTagCompound compoundTag;
+			try
 			{
-				t.stackSize = this.tagCompound.getInteger( itemSlotCount[x] );
+				compoundTag = this.tagCompound.getCompoundTag( itemSlots[x] );
+			}
+			catch( Throwable ex )
+			{
+				if( AEConfig.instance.removeCrashingItemsOnLoad )
+				{
+					AELog.warn( ex, "Removing item from storage cell slot " + itemSlots[x] + " because it's NBT could not be loaded." );
+					continue;
+				}
+				throw ex;
+			}
 
-				if( t.stackSize > 0 )
+			// Now load the item stack
+			final ItemStack t;
+			try
+			{
+				t = ItemStack.loadItemStackFromNBT( compoundTag );
+				if( t == null )
+				{
+					AELog.warn( "Removing item " + compoundTag + " from storage cell because the associated item type couldn't be found." );
+					continue;
+				}
+			}
+			catch( Throwable ex )
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Catch Exception instead of Throwable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1181)